@xen-orchestra/log/logger.test.js

```diff
@@ -0,0 +1,154 @@
+'use strict'
+
+const { beforeEach, describe, it } = require('node:test')
+const { configure } = require('@xen-orchestra/log/configure')
+const { createCaptureTransport } = require('@xen-orchestra/log/capture')
+const { createLogger } = require('@xen-orchestra/log')
+const assert = require('node:assert/strict')
+
+describe('logger', () => {
+  const logger = createLogger('test-logger')
+
+  const logsTransportDefault = []
+  const transportTest = message => {
+    logsTransportDefault.push(message)
+  }
+
+  const logsTransportFatal = []
+  const transportFatal = message => {
+    logsTransportFatal.push(message)
+  }
+
+  beforeEach(() => {
+    configure(
+      createCaptureTransport([
+        {
+          level: 'fatal',
+          transport: transportFatal,
+        },
+        {
+          level: 'debug',
+          transport: transportTest,
+        },
+      ])
+    )
+
+    logsTransportDefault.length = 0
+    logsTransportFatal.length = 0
+  })
+
+  it('should log test with all its attributes', () => {
+    const expected1 = {
+      level: 20,
+      namespace: 'test-logger',
+      message: 'test 1 with all its attributes',
+      time: new Date(),
+    }
+    logger.debug(expected1.message)
+    assert.equal(logsTransportDefault[0].message, expected1.message)
+    assert.equal(typeof logsTransportDefault[0].level, 'number', `level attribute should exist`)
+    assert.equal(typeof logsTransportDefault[0].namespace, 'string', `namespace attribute should exist`)
+    assert.equal(logsTransportDefault[0].time instanceof Date, true, 'time attribute should exist')
+  })
+  it('should log test 1 and 2', () => {
```

Comment:
What feature does this case test that the the previous one did not?

Reply:
It test the "test transport" can contain many logs, and not only one constantly overwriten. Needed by the captureLog test after when we test multiple logs.